src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -0,0 +1,237 @@
+            {
+                if (!_disposedValue)
+                {
+                    if (disposing)
+                    {
+                        _topLevelNullabilityMap.Free();
+                    }
+
+                    _disposedValue = true;
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(true);
+            }
+
+            public override BoundNode VisitAnonymousObjectCreationExpression(BoundAnonymousObjectCreationExpression node)
+            {
+
+                if (node.Declarations.Length != node.Arguments.Length)
+                {
+                    // https://github.com/dotnet/roslyn/issues/24018 implicit anonymous properties are not currently handled,
+                    // so skip verification on the declarations for now and only verify the AnonymousObjectCreation and the
+                    // arguments.
+                    foreach (var arg in node.Arguments)
+                    {
+                        Visit(arg);
+                    }
+                    return null;
+                }
+                else
+                {
+                    return base.VisitAnonymousObjectCreationExpression(node);
+                }
+            }
+
+            public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationExpression node)
+            {
+                // If the delegate creation was resolved to a single static method, we do not examine the
+                // receiver of the method group as it's completely unused.
+                if (node.Argument is BoundMethodGroup group)
+                {
+                    VerifyNode(group);
+                    if (node.MethodOpt?.IsStatic == false)
+                    {
+                        Visit(group.ReceiverOpt);
+                    }
+
+                    return null;
+                }
+                else
+                {
+                    return base.VisitDelegateCreationExpression(node);
+                }
+            }
+
+            public override BoundNode VisitNewT(BoundNewT node)
+            {
+                // https://github.com/dotnet/roslyn/issues/33387 We're not currently
+                // examining child nodes correctly
+                if (node.InitializerExpressionOpt != null)
+                {
+                    VerifyNode(node.InitializerExpressionOpt, overrideSkippedExpression: true);
+                }
+
+                return null;
+            }
+
+            public override BoundNode VisitBadExpression(BoundBadExpression node)
+            {
+                // Regardless of what the BadExpression is, we need to add all of its direct children to the visited map. They
+                // could be things like object initializers (see New_01.F1).
+                foreach (var child in node.ChildBoundNodes)
+                {
+                    if (!s_skippedExpression.Contains(child.Kind))
```

Comment:
>s_skippedExpression [](start = 25, length = 19)

Consider replacing `s_skippedExpression` with a `switch` statement.

Reply:
I'm specifically not doing that because there are multiple places we need that info.

---
In reply to: [260070456](https://github.com/dotnet/roslyn/pull/33591#discussion_r260070456) [](ancestors = 260070456)